internal/controller/httproute_controller.go

```diff
@@ -244,6 +280,76 @@ func (r *HTTPRouteReconciler) listHTTPRoutesForGateway(ctx context.Context, obj
 	return requests
 }
 
+func (r *HTTPRouteReconciler) listHTTPRouteByHTTPRoutePolicy(ctx context.Context, obj client.Object) (requests []reconcile.Request) {
+	httpRoutePolicy, ok := obj.(*v1alpha1.HTTPRoutePolicy)
+	if !ok {
+		r.Log.Error(fmt.Errorf("unexpected object type"), "failed to convert object to HTTPRoutePolicy")
+		return nil
+	}
+
+	var keys = make(map[targetRefKey]struct{})
+	for _, ref := range httpRoutePolicy.Spec.TargetRefs {
+		if ref.Kind == "HTTPRoute" {
+			key := targetRefKey{
+				Group:     gatewayv1.GroupName,
+				Namespace: gatewayv1.Namespace(obj.GetNamespace()),
+				Name:      ref.Name,
+			}
+			if ref.SectionName != nil {
+				key.SectionName = *ref.SectionName
+			}
+			keys[key] = struct{}{}
+		}
+	}
+	for key := range keys {
+		var httpRoute gatewayv1.HTTPRoute
+		if err := r.Get(ctx, client.ObjectKey{Namespace: string(key.Namespace), Name: string(key.Name)}, &httpRoute); err != nil {
+			r.Log.Error(err, "failed to get HTTPRoute by HTTPRoutePolicy targetRef", "namespace", key.Namespace, "name", key.Name)
+			continue
+		}
+		if key.SectionName != "" {
+			var matchRuleName bool
+			for _, rule := range httpRoute.Spec.Rules {
+				if rule.Name != nil && *rule.Name == key.SectionName {
+					matchRuleName = true
+					break
+				}
+			}
+			if !matchRuleName {
+				r.Log.Error(errors.Errorf("failed to get HTTPRoute rule by HTTPRoutePolicy targetRef"), "namespace", key.Namespace, "name", key.Name, "sectionName", key.SectionName)
+				continue
+			}
+		}
+		requests = append(requests, reconcile.Request{
+			NamespacedName: types.NamespacedName{
+				Namespace: string(key.Namespace),
+				Name:      string(key.Name),
+			},
+		})
+	}
+
+	return requests
+}
+
+func (r *HTTPRouteReconciler) listHTTPRouteForGenericEvent(ctx context.Context, obj client.Object) []reconcile.Request {
+	switch v := obj.(type) {
+	case *v1alpha1.HTTPRoutePolicy:
+		var (
+			namespacedNames = make(map[types.NamespacedName]struct{})
+			requests        []reconcile.Request
+		)
+		for _, ref := range v.Spec.TargetRefs {
+			namespacedName := types.NamespacedName{Namespace: v.GetNamespace(), Name: string(ref.Name)}
```

Comment:
1. Need to check for duplicates as there are different sectionNames
2. It is necessary to check if the httproute exists, otherwise it will trigger the deletion operation of the data plane.

Reply:
resolved